code/site/components/com_pages/view/html.php

```diff
@@ -183,47 +186,38 @@ public function getMetadata()
         return $metadata;
     }
 
-    public function getRoute($route = '', $fqr = true, $escape = true)
+    public function getRoute($page = '', $query = array(), $escape = false)
```

Comment:
@johanjanssens I feel that something is missing here. All is well as long as a $page is provided. However I see things not working as expected if route() is called without args. On this last case I would expect to get a route for the current page (states included). In this case $page should default to $this->getPage(). Thoughts?

Reply:
Changed the code to allow for an empty route and use the active page to generate it. This scenario will not happen much, but is now also supported.